plasoscaffolder/frontend/frontend.py

```diff
@@ -0,0 +1,373 @@
+# !/usr/bin/python
+# -*- coding: utf-8 -*-
+"""The scaffolder frontend."""
+import re
+
+from typing import Dict
+from typing import List
+from typing import Type
+
+from plasoscaffolder.frontend import output_handler
+from plasoscaffolder.helpers import git
+from plasoscaffolder.lib import engine
+from plasoscaffolder.lib import errors
+
+from plasoscaffolder.definitions import interface as definition_interface
+from plasoscaffolder.definitions import manager as definition_manager
+from plasoscaffolder.scaffolders import interface as scaffolder_interface
+from plasoscaffolder.scaffolders import manager as scaffolder_manager
+
+
+class ScaffolderFrontend:
+  """A frontend implementation for the scaffolder project.
+
+  Attributes:
+    OUTPUT_HANDLER (output_handler.BaseOutputHandler: output handler
+        that is used to request and read input from end user.
+  """
+
+  OUTPUT_HANDLER = output_handler.BaseOutputHandler()
+
+  @classmethod
+  def _AskDictQuestion(
+      cls, question: scaffolder_interface.BaseQuestion) -> Dict[str, Type]:
```

Comment:
It looks like this needs to take a DictQuestion, not a BaseQuestion - please fix.

Reply:
This also seems to return a dict of [str, str].